velox/exec/fuzzer/AsyncDataCacheFuzzer.cpp

```diff
@@ -0,0 +1,354 @@
+  if (fileIds_.empty()) {
+    for (auto i = 0; i < FLAGS_num_files; ++i) {
+      auto fileName = fmt::format("{}/file_{}", tempDirectory_->getPath(), i);
+
+      const size_t fileSize = boost::random::uniform_int_distribution<uint64_t>(
+          FLAGS_min_file_bytes, FLAGS_max_file_bytes)(rng_);
+      auto writeFile = LocalWriteFile(fileName);
+      size_t writtenSize = 0;
+      int32_t offset = 0;
+      while (writtenSize < fileSize) {
+        size_t chunkSize = std::min(
+            fileSize - writtenSize,
+            size_t(4 << 20)); // Write in chunks of 4MB
+        auto buffer = folly::IOBuf::create(chunkSize);
+        buffer->append(chunkSize);
+        // Fill buffer with data.
+        std::generate_n(
+            buffer->writableData(), chunkSize, [&offset]() -> uint8_t {
+              return static_cast<uint8_t>(offset++ % 256);
+            });
+        writeFile.append(std::move(buffer));
+        writtenSize += chunkSize;
+      }
+      writeFile.close();
+
+      fileIds_.emplace_back(fileIds(), fileName);
+      readFiles_.emplace_back(std::make_shared<LocalReadFile>(fileName));
+      fileSizes_.emplace_back(fileSize);
+    }
+  }
+}
+
+void AsyncDataCacheFuzzer::initializeCache() {
+  std::unique_ptr<SsdCache> ssdCache;
+  if (FLAGS_ssd_cache_bytes > 0) {
+    SsdCache::Config config(
+        fmt::format("{}/cache", tempDirectory_->getPath()),
+        FLAGS_ssd_cache_bytes,
+        FLAGS_num_shards,
+        executor(),
+        FLAGS_ssd_checkpoint_interval_bytes,
+        false,
+        FLAGS_enable_checksum,
+        FLAGS_enable_read_verification);
+    ssdCache = std::make_unique<SsdCache>(config);
+  }
+
+  memory::MemoryManagerOptions options;
+  options.useMmapAllocator = true;
+  options.allocatorCapacity = FLAGS_memory_cache_bytes;
+  options.arbitratorCapacity = FLAGS_memory_cache_bytes;
+  options.arbitratorReservedCapacity = 0;
+  options.trackDefaultUsage = true;
+  manager_ = std::make_unique<memory::MemoryManager>(options);
+  allocator_ = dynamic_cast<memory::MmapAllocator*>(manager_->allocator());
+
+  // TODO: Test different ssd write behaviors with AsyncDataCache::Options.
+  cache_ = AsyncDataCache::create(allocator_, std::move(ssdCache), {});
+}
+
+void AsyncDataCacheFuzzer::initializeInputs() {
+  auto readOptions = io::ReaderOptions(pool_.get());
+  auto tracker = std::make_shared<ScanTracker>(
+      "testTracker", nullptr, io::ReaderOptions::kDefaultLoadQuantum);
+  auto ioStats = std::make_shared<IoStatistics>();
+  inputs_.reserve(FLAGS_num_files);
+  for (auto i = 0; i < FLAGS_num_files; ++i) {
+    groupIds_.emplace_back(
+        fileIds(), fmt::format("group{}", i / FLAGS_num_files_in_group));
+    inputs_.emplace_back(std::make_unique<CachedBufferedInput>(
+        readFiles_[i], // NOLINT
+        MetricsLog::voidLog(),
+        fileIds_[i].id(), // NOLINT
+        cache_.get(),
+        tracker,
+        groupIds_[i].id(),
+        ioStats,
+        executor_.get(),
```

Comment:
Test with and without the executor? Thanks!

Reply:
executor of SsdCache cannot be nullptr.